From: Zhong LuFan <lufanzhong@gmail.com>
Subject: [PATCH] ARM: dts: BCM5301X: Add basic DT for PHICOMM K3

This router has BCM4709C0, 128MB NAND flash (MX30LF1G18AC-TI),
and 512MB memory, with 3 x LAN and 1 x WAN. WL chips are
BCM4366C0 x 2. The router has a small LCD and 3 capactive keys
driven by a PIC microcontroller, which is in turn wired to
UART1 of main board.

Everything except the LCD and wireless works.

Flash:
Enter CFE
CFE Brush Excess LEDE, you can get it from here: https://drive.google.com/file/d/0ByNlqhxfFBArb2owR05VUlVpN0E
Well, now you can brush into the trx file for the LEDE.

The founder of the patch is: Tian Hao <haotia@gmail.com>

Signed-Off-By: Zhong LuFan <lufanzhong@gmail.com>

--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile 
@@ -100,6 +100,7 @@ dtb-$(CONFIG_ARCH_BCM_5301X) += \
 	bcm47094-luxul-xbr-4500.dtb \
 	bcm47094-luxul-xwr-3100.dtb \
 	bcm47094-netgear-r8500.dtb \
+	bcm47094-phicomm-k3.dtb \
 	bcm94708.dtb \
 	bcm94709.dtb \
 	bcm953012er.dtb \
--- /dev/null
+++ b/arch/arm/boot/dts/bcm47094-phicomm-k3.dts
@@ -0,0 +1,37 @@
+/*
+* Copyright (C) 2017 Hamster Tian <haotia@gmail.com>
+*/
+
+/dts-v1/;
+
+#include "bcm47094.dtsi"
+#include "bcm5301x-nand-cs0-bch4.dtsi"
+
+/ {
+	compatible = "phicomm,k3", "brcm,bcm47094", "brcm,bcm4708";
+	model = "PHICOMM K3";
+	
+	chosen {
+		bootargs = "console=ttyS0,115200";
+    	};
+	memory {
+		reg = <0x00000000 0x08000000
+			0x88000000 0x18000000>;
+ 	};
+ 
+ 	gpio-keys {
+		compatible = "gpio-keys";
+		#address-cells = <1>;
+		#size-cells = <0>;
+ 
+ 		restart {
+ 			label = "Reset";
+ 			linux,code = <KEY_RESTART>;
+ 			gpios = <&chipcommon 17 GPIO_ACTIVE_LOW>;
+ 		};
+ 	};
+ };
+ 
+ &uart1 {
+ 	status = "okay";
+ };
